### PR TITLE
Version 0.25

### PR DESCRIPTION
**Pull request by locusf**
* Device lock
* Sim panel has reasonable buttons
* Label renames as per @Sfiet_Konstantin

### CODE REVIEW

File: src/qml/DeviceLock.qml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            readOnly: true
+            echoMode: TextInput.PasswordEchoOnEdit
+        }
+
+        GridLayout {
+            height: parent.height
+            width: parent.width
+            columns: 3
+            Repeater {
+                model: [0,1,2,3,4,5,6,7,8,9,10,11]
```
Comment (SfietKonstantin) [inline src/qml/DeviceLock.qml]: That's rpetty bad, especially when you handle it afterwards as Ca or OK

File: src/qml/Lockscreen.qml [hunk 2]
```diff
@@ -6,12 +6,21 @@ Image {
...
             top: parent.top
             left: parent.left
             right: parent.right
         }
     }
+    DeviceLock {
+        id: deviceLockUI
+        anchors.fill: parent
+        enabled: deviceLock.state === 1
+        opacity: deviceLock.state === 1 ? 1 : 0
```
Comment (SfietKonstantin) [inline src/qml/Lockscreen.qml]: Use visible

**Comment by SfietKonstantin** [comment]
Just wondering, can't you export the SIM panel and share it with lock screen code ?
Do a Dialer component ?

File: src/qml/DeviceLock.qml (hunk 1 above)
Comment (locusf) [inline src/qml/DeviceLock.qml]: Yeah, I don't really know how to make a mixed model like that, do you have any suggestions?

File: src/qml/Lockscreen.qml (hunk 2 above)
Comment (locusf) [inline src/qml/Lockscreen.qml]: Okay will fix as visible.

**Comment by locusf** [comment]
Hmm maybe, but the Dialer isn't quite ready yet. A full screen SIM PIN unlocker is the best solution but I don't know how to get there yet.

File: src/qml/DeviceLock.qml (hunk 1 above)
Comment (SfietKonstantin) [inline src/qml/DeviceLock.qml]: What happens if you use strings ?

Comment (locusf) [inline src/qml/DeviceLock.qml]: Yeah strings work, didn't know that, thanks!

**Comment by SfietKonstantin** [comment]
What's the difference in implementation between dialer and lockscreen ATM. I have seen that dialer uses layout while lockscreen uses grid.

**Comment by locusf** [comment]
Both use grid afaik.

**Comment by locusf** [comment]
Is this lgtm?

**Pull request merged**